src/main/kotlin/gg/essential/elementa/markdown/MarkdownComponent.kt

```diff
@@ -133,33 +134,33 @@ class MarkdownComponent @JvmOverloads constructor(
         setHeight((currY - baseY).coerceAtMost(maxHeight.getHeight(this)).pixels())
     }
 
-    override fun afterInitialization() {
-        reparse()
-        layout()
-        lastValues = constraintValues()
-    }
+    override fun animationFrame() {
+        super.animationFrame()
 
-    override fun draw(matrixStack: UMatrixStack) {
-        if (!isInitialized) {
-            isInitialized = true
-            afterInitialization()
+        if (!initialLayout) {
+            initialLayout = true
+            reparse()
+            layout()
+            lastValues = constraintValues()
         }
 
-        beforeChildrenDraw(matrixStack)
-
         // Re-layout if important constraint values have changed
         val currentValues = constraintValues()
         if (currentValues != lastValues)
             layout()
         lastValues = currentValues
+    }
+
+    override fun draw(matrixStack: UMatrixStack) {
+        beforeDrawCompat(matrixStack)
```

Comment:
Why wouldn't this just be beforedraw?

Reply:
Force of habit. It can just be beforeDraw.